syntax = "proto3";

package permission;

message PermissionRequest {
    string session_token = 1;
    string service = 2;
    string entity = 3;
    string action = 4;
}

message PermissionResponse {
    bool is_access = 1;
    string message = 2;
    int32 code = 3;
}

service PermissionService {
    rpc ValidatePermission(PermissionRequest) returns (PermissionResponse);
}
